src/mesh/data/gridfromoptions.cxx

```diff
@@ -91,45 +91,45 @@ bool GridFromOptions::get(Mesh *m, vector<int> &var, const string &name, int len
   if(!get(m, ival, name))
     return false;
 
-  var.resize(len);
-
-  for(int i=0; i<var.size(); i++)
-    var[i] = ival;
+  var.resize(len, ival);
 
   return true;
 }
 
-bool GridFromOptions::get(Mesh *m, vector<BoutReal> &var, const string &name, int len, int offset, GridDataSource::Direction dir) {
-  
+bool GridFromOptions::get(Mesh *m, vector<BoutReal> &var, const string &name, int len,
+                          int offset, GridDataSource::Direction dir) {
+
   if(!hasVar(name)) {
     return false;
   }
-  
-  // Fetch expression as a string 
+
+  // Fetch expression as a string
   std::string expr;
   options->get(name, expr, "0");
-  
+
   // Parse, and evaluate with x,y,z,t = 0
   FieldGenerator* gen = FieldFactory::get()->parse(expr, options);
 
   var.resize(len);
-  
+
   switch(dir) {
   case GridDataSource::X: {
-    for(int x=0;x<var.size();x++){
```

Comment:
I assume the compiler warnung was comparison of signed and unsigned int?
Wouldn't it make more sense to use an unsigned int for `x` and `len`?